Python/Product/PythonTools/PythonTools/Refactoring/ExtractMethodRequestView.cs

```diff
@@ -256,9 +257,14 @@ public ExtractMethodRequestView(IServiceProvider serviceProvider, ExtractedMetho
         private void UpdatePreview() {
             var info = GetRequest();
             if (info != null) {
-                _previewer.GetExtractionResult(info).ContinueWith(
-                    x => PreviewText = x?.WaitOrDefault(1000)?.methodBody ?? Strings.ExtractMethod_FailedToGetPreview
-                ).DoNotWait();
+                _previewer.GetExtractionResult(info).ContinueWith(t => {
+                    try {
+                        PreviewText = t.Result.methodBody;
```

Comment:
is this the thing that used to return null when user selected code that isn't valid to be extracted and then crashed?  If so you may want to keep the r.Result?.methodBody.  I know you catch Exception, but failing to get preview due to a normal scenario doesn't need to Debug.Fail.